lib/vectorsearch/pinecone.rb

```diff
@@ -26,13 +26,14 @@ def initialize(environment:, api_key:, index_name:, llm:, llm_api_key:)
     # Add a list of texts to the index
     # @param texts [Array] The list of texts to add
     # @param namespace [String] The namespace to add the texts to
+    # @param metadata [Hash] The metadata to use for the texts
     # @return [Hash] The response from the server
-    def add_texts(texts:, namespace: "")
+    def add_texts(texts:, namespace: "", metadata: nil)
       vectors = texts.map do |text|
         {
           # TODO: Allows passing in your own IDs
           id: SecureRandom.uuid,
-          metadata: {content: text},
+          metadata: metadata || {content: text},
```

Comment:
I can't recall why `{content: text}` was previously passed in. Is it a required attribute?

Reply:
Based on Pinecone documentation, it seems like this can be any key:value metadata that we want to use to provide extra context around upserted vector values, and seemingly `content` is not a required or 'magic' key.

https://docs.pinecone.io/docs/insert-data#inserting-vectors-with-metadata 
https://docs.pinecone.io/docs/metadata-filtering
https://docs.pinecone.io/reference/upsert